botocore/parsers.py

```diff
@@ -250,7 +250,13 @@ def parse(self, response, shape):
         if isinstance(parsed, dict):
             response_metadata = parsed.get('ResponseMetadata', {})
             response_metadata['HTTPStatusCode'] = response['status_code']
-            response_metadata['HTTPHeaders'] = dict(response['headers'])
+            http_headers = {}
+            # Ensure that the http header keys are all lower cased. Older
+            # versions of urllib3 (< 1.11) would unintentionally do this for us
+            # (see urllib3#633). We need to do this conversion manually now.
+            for header in response['headers']:
```

Comment:
I'm not a fan of this.  It doesn't seem like it's the parsers responsibility to handle the lower casing, which I think is further hinted at by the fact that we're changing our parser because our http client changed (which I would not expect).

What was the reason for not fixing this in the custom headers dictionary class we made?

Reply:
We're not fixing this in the custom headers dictionary because the previous headers object wasn't responsible for doing this either. This change in behavior would have happened even if we just upgraded requests. This behavior was never intentional on our part or urllib3s part, but is now part of our public contract for what we return. The parsers are what converts the response's header object into a `dict` and is what ultimately dictates the final format and structure of what we return from client operations.

Reply:
Ok, I think that sounds reasonable.  Could you have this off to a utils function?  Some of the other dict manipulation functions are already in utils (e.g `merge_dicts`).

Reply:
Sure, sounds reasonable.